UnityProject/Assets/Scripts/Items/Implants/Organs/BrainLaws.cs

```diff
@@ -0,0 +1,465 @@
+			ResetLaws(isPurge);
+			Chat.AddActionMsgToChat(interaction.Performer,
+				$"You {(isPurge ? "purge" : "reset")} all of {gameObject.ExpensiveName()}'s laws",
+				$"{interaction.Performer.ExpensiveName()} {(isPurge ? "purges" : "resets")} all of {gameObject.ExpensiveName()}'s laws");
+			return;
+		}
+
+		if (lawFromModule.Count == 0)
+		{
+			Chat.AddExamineMsgFromServer(interaction.Performer, "No laws to upload");
+			return;
+		}
+
+		//If we are only adding core laws then we must mean to remove old core laws
+		//This means we are assuming that the law set must only have core laws if it is to replace the old laws fully
+		var notOnlyCoreLaws = false;
+
+		foreach (var law in lawFromModule)
+		{
+			if (law.Key != AiPlayer.LawOrder.Core)
+			{
+				notOnlyCoreLaws = true;
+				break;
+			}
+		}
+
+		SetLaws(lawFromModule, true, notOnlyCoreLaws);
+
+		Chat.AddActionMsgToChat(interaction.Performer, $"You change {gameObject.ExpensiveName()} laws",
+			$"{interaction.Performer.ExpensiveName()} changes {gameObject.ExpensiveName()} laws");
+	}
+
+	//Add one law
+	//Wont allow more than one traitor law
+	[Server]
+	public void AddLaw(string newLaw, AiPlayer.LawOrder order, bool init = false)
+	{
+		if (aiLaws.ContainsKey(order) == false)
+		{
+			aiLaws.Add(order, new List<string>() {newLaw});
+		}
+		else
+		{
+			if (order == AiPlayer.LawOrder.Traitor && aiLaws[order].Count > 0)
+			{
+				//Can only have one traitor law
+				return;
+			}
+
+			if (aiLaws[order].Contains(newLaw))
+			{
+				//Cant add the same law with the same string more than once
+				return;
+			}
+
+			aiLaws[order].Add(newLaw);
+		}
+
+		//Dont spam client on init
+		if (init) return;
+
+		Chat.AddExamineMsgFromServer(gameObject, "Your Laws Have Been Updated!");
+
+		//Tell player to open law screen so they dont miss that their laws have changed
+		ServerUpdateClientLaws();
+	}
+
+	//Set a new list of laws, used mainly to set new core laws, can remove core laws if parameter set true
+	//Wont replace hacked. ion laws and freeform can be replaced if parameters set to false
+	[Server]
+	public void SetLaws(Dictionary<AiPlayer.LawOrder, List<string>> newLaws, bool keepIonLaws = true,
+		bool keepCoreLaws = false,
+		bool keepFreeform = true)
+	{
+		foreach (var lawGroups in aiLaws)
+		{
+			if (lawGroups.Key == AiPlayer.LawOrder.Traitor)
+			{
+				TryAddLaw(AiPlayer.LawOrder.Traitor);
+			}
+
+			if (keepIonLaws && lawGroups.Key == AiPlayer.LawOrder.Hacked)
+			{
+				TryAddLaw(AiPlayer.LawOrder.Hacked);
+			}
+
+			if (keepIonLaws && lawGroups.Key == AiPlayer.LawOrder.IonStorm)
+			{
+				TryAddLaw(AiPlayer.LawOrder.IonStorm);
+			}
+
+			if (keepCoreLaws && lawGroups.Key == AiPlayer.LawOrder.Core)
+			{
+				TryAddLaw(AiPlayer.LawOrder.Core);
+			}
+
+			if (keepFreeform && lawGroups.Key == AiPlayer.LawOrder.Freeform)
+			{
+				TryAddLaw(AiPlayer.LawOrder.Freeform);
+			}
+
+			void TryAddLaw(AiPlayer.LawOrder order)
+			{
```

Comment:
actually, due to how big this is; its probably best that you put it in its own function outside